src/ImageSharp/PixelFormats/PixelImplementations/PixelOperations/Generated/_Common.ttinclude

```diff
@@ -8,19 +8,85 @@
 
 // <auto-generated />
 
-using SixLabors.ImageSharp.PixelFormats.Utils;
 using System;
-using System.Buffers;
 using System.Numerics;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
+using SixLabors.ImageSharp.Formats;
+using SixLabors.ImageSharp.PixelFormats.Utils;
 <#+
-    static readonly string[] CommonPixelTypes = { "Argb32", "Bgr24", "Bgra32", "L8", "L16", "La16", "La32", "Rgb24", "Rgba32", "Rgb48", "Rgba64", "Bgra5551" };
-
-    static readonly string[] OptimizedPixelTypes = { "Rgba32", "Argb32", "Bgra32", "Rgb24", "Bgr24" };
+    private static readonly string[] UnassociatedAlphaPixelTypes =
+    {
+        "A8",
+        "Argb32",
+        "Bgra32",
+        "Bgra4444",
+        "Bgra5551",
+        "Byte4",
+        "HalfVector4",
+        "La16",
+        "La32",
+        "NormalizedByte4",
+        "NormalizedShort4",
+        "Rgba1010102",
+        "Rgba32",
+        "Rgba64",
+        "RgbaVector",
+        "Short4"
+    };
+
+    private static readonly string[] AssociatedAlphaPixelTypes = Array.Empty<string>();
+
+    private static readonly string[] CommonPixelTypes =
+    {
+        "Argb32",
+        "Bgr24",
+        "Bgra32",
+        "L8",
+        "L16",
+        "La16",
+        "La32",
+        "Rgb24",
+        "Rgba32",
+        "Rgb48",
+        "Rgba64",
+        "Bgra5551"
+    };
+
+    private static readonly string[] OptimizedPixelTypes =
+    {
+        "Rgba32",
+        "Argb32",
+        "Bgra32",
+        "Rgb24",
+        "Bgr24"
+    };
+
+    // Types with Rgba32-combatable to/from Vector4 conversion
+    private static readonly string[] Rgba32CompatibleTypes =
+    {
+        "Argb32",
+        "Bgra32",
+        "Rgb24",
+        "Bgr24"
+    };
 
-    // Types with Rgba32-combatible to/from Vector4 conversion
-    static readonly string[] Rgba32CompatibleTypes = { "Argb32", "Bgra32", "Rgb24", "Bgr24" };
+    void GeneratePixelTypeInfo(string pixelType)
+    {
+        string alpha = "PixelAlphaRepresentation.None";
```

Comment:
When we add a new pixel type with a `.tt`, we should be careful and include it in `UnassociatedAlphaPixelTypes` when it has an alpha, otherwise `PixelTypeInfo.AlphaRepresentation` will be incorrectly report `None`.

Reply:
Yeah... My thinking was since this was only used when we are adding via `.tt` we have to check the various lists anyway for correctness.

Reply:
I've moved the code to an explicit declaration